caddyfile/parse.go

```diff
@@ -433,3 +457,41 @@ type ServerBlock struct {
 	Keys   []string
 	Tokens map[string][]Token
 }
+
+func (p *parser) isSnippet() (bool, string) {
+	keys := p.block.Keys
+	// A snippet block is a single key with parens. Nothing else qualifies.
+	if len(keys) == 1 && strings.HasPrefix(keys[0], "(") && strings.HasSuffix(keys[0], ")") {
+		return true, strings.TrimSuffix(keys[0][1:], ")")
```

Comment:
Why TrimSuffix instead of just `[1:len(keys[0])-1]`?

Reply:
Can we explicitly check here that the snippet name inside the parens () does not include a `.` (dot) this will avoid 1 possible route to confusion.

Reply:
What is confusing about it?

Reply:
I just think it would be good to avoid

(example.com) {
   log example.log "{common}"
}

as a snippet

Reply:
Hmm. Well, I'm curious to see if that will actually be a problem. How about we wait and find out before we add a restriction...

Reply:
I'm not sure I understand why it shouldn't be added as a restriction? It needs to be added from the beginning or not at all, otherwise you will have the potential to break installations.  The only legitimate use I could see would be if someone wished to create a snippet 

     (snippet for all subdomains of example.com) {
          tls off
       }

I wonder how likely that is.  I would suggest it is easier to remove the restriction if  people feel it gets in the way rather than add it in at a later point.

Reply:
Hmm, well, I see what you mean... (I think this change requires that the name in parens is only one word, and I assume your example is just an example in that sense) - what do you think @captncraig ?